File: main-server-admin/src/database/MySQL.js
import mysql from 'mysql2';

class MySql {
    constructor(config) {
        this.pool = mysql.createPool(config);
    }

    getConnection() {
        return new Promise((resolve, reject) => {
            this.pool.getConnection((err, connection) => {
                if (err) return reject(err)
                resolve(connection);
            });
        });
    }

    startTransaction(connection) {
        return new Promise((resolve, reject) => {
            connection.beginTransaction(err => {
                if (err) return reject(err);
                resolve();
            });
        });
    }
    
    releaseTransaction(connection) {
        return connection.release();
    }

    commitTransaction(connection) {
        return new Promise((resolve, reject) => {
            connection.commit(err => {
                if (err) return reject(err);
                resolve();
            })
        })
    }

    rollbackTransaction(connection) {
        return new Promise((resolve, reject) => {
            connection.rollback(err => {
                if (err) return reject(err);
                resolve();
            })
        })
    }

    query(sql, params, connection) {
        const executor = connection || this.pool;
        return new Promise((resolve, reject) => {
            executor.query(sql, params, (err, result) => {
                if (err) {
                    return reject(err);
                }
                resolve(result);
            });
        });
    }

    close() {
        return new Promise((resolve, reject) => {
            this.pool.end(err => {
                if (err) reject(err);
                resolve("Connection closed");
            })
        })
    }
}

export default MySql

File: telegram-bot/src/database/MySql.py
import atexit
import signal

import mysql.connector as connector
from mysql.connector import pooling, errorcode, Error
from typing import Tuple, Dict, Union

class MySql:
    def __init__(self, **config: dict) -> None:
        try:
            self.pool = pooling.MySQLConnectionPool(
                pool_reset_session=True,
                **config
            )
            atexit.register(self.__shutdown)
            signal.signal(signal.SIGINT, self.__handle_exit)
            signal.signal(signal.SIGTERM, self.__handle_exit)
        except connector.Error as err:
            if err.errno == errorcode.ER_ACCESS_DENIED_ERROR:
                print("Something is wrong with your user name or passmord")
            elif err.errno == errorcode.ER_BAD_DB_ERROR:
                print("Database does not exist")
            else:
                print("Error connection to database: ", err)
    
    def __get_connection(self):
        try:
            conn = self.pool.get_connection()
            cursor = conn.cursor()
            return conn, cursor
        except Error as err:
            print("Error getting connection from pool:", err)
            raise err

    def __shutdown(self):
        self.pool = None

    def query(self, sql: str, params: dict=None) -> Union[Tuple, Dict, None]:
        conn, cursor = self.__get_connection()
        try:
            cursor.execute(operation=sql, params=params)
            return cursor.fetchone()
        except connector.Error as err:
            print("Failed to complete the query: ", err)
            raise err
        finally:
            cursor.close()
            conn.close()

    def execute(self, sql: str, params: dict=None) -> None:
        conn, cursor = self.__get_connection()
        try:
            cursor.execute(operation=sql, params=params)
            conn.commit()
        except connector.Error as err:
            print("Failed to complete the query: ", err)
            raise err
        finally:
            cursor.close()
            conn.close()
        
    def __handle_exit(self, signum, frame):
        self.__shutdown()